packages/notte-agent/src/notte_agent/falco/agent.py

```
@@ -301,6 +303,30 @@ async def _run(self, request: AgentRunRequest) -> AgentResponse:
+
+        if request.upload_dir is not None:
+            # To consider: should upload directory paths be validated to prevent escape?
+            # ex. don't allow upload_dir paths with ".." or "~" etc.
+
+            upload_dir = Path(request.upload_dir)
+            all_files = [
+                str(p)
+                for p in upload_dir.rglob("*")
+                if p.is_file()
+                and not p.name.startswith(".")
+                and not any(part.startswith(".") for part in p.parts[len(upload_dir.parts) :])
```

Flake8 (7.2.0)

[error] whitespace before ':'

(E203)